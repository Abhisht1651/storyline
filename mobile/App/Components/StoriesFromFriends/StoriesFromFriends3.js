// @flow

import React from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import Images from '../../Themes/Images'
import styles from '../Styles/StoriesFromFriendsStyles/StoriesFromFriends3Style'

type Props = {
  image1: Image,
  image2: Image,
  name: string,
  onPress: () => {}
}

const StoriesFromFriends3 = (props: Props) => {
  const { image1, image2, name, onPress } = props

  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <TouchableOpacity style={styles.profilePicture} activeOpacity={1}>
        <Image source={image1} style={styles.image} />
      </TouchableOpacity>
      <View style={styles.TriangleShapeCSS} />
      <TouchableOpacity activeOpacity={1} style={styles.storyIcon}>
        <View style={styles.smallImageContainer}>
          <Image source={image2} style={styles.storyImage} />
          <Image source={image2} style={styles.secondaryImage} />
        </View>
      </TouchableOpacity>
      <Text style={styles.text}>{name}</Text>
    </TouchableOpacity>
  )
}

StoriesFromFriends3.defaultProps = {
  image1: Images.flagPicture,
  image2: Images.profilePicture4,
  name: 'Carlota'
}

export default StoriesFromFriends3
