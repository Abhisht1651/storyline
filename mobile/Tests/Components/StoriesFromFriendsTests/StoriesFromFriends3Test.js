import { shallow } from 'enzyme'
import React from 'react'
import StoriesFromFriends3 from '../../../App/Components/StoriesFromFriends/StoriesFromFriends3'

describe('StoriesFromFriends3', () => {
  it('should match to snapshot', () => {
    const component = shallow(<StoriesFromFriends3 />)
    expect(component).toMatchSnapshot()
  })
  it('should render Image', () => {
    const component = shallow(<StoriesFromFriends3 />)
    expect(component.find('Image')).toHaveLength(3)
  })
})
