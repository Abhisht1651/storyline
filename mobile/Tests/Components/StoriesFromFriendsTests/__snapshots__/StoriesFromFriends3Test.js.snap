// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoriesFromFriends3 should match to snapshot 1`] = `
<ForwardRef
  style={
    Object {
      "alignItems": "center",
      "height": 211.92857142857142,
      "justifyContent": "space-between",
      "width": 118.28571428571428,
    }
  }
>
  <ForwardRef
    activeOpacity={1}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#DCDCE9",
        "borderRadius": 24.642857142857142,
        "borderWidth": 2,
        "height": 152.78571428571428,
        "justifyContent": "center",
        "width": 118.28571428571428,
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "borderColor": "white",
          "borderRadius": 21.357142857142858,
          "borderWidth": 2,
          "height": 146.21428571428572,
          "width": 111.71428571428571,
        }
      }
    />
  </ForwardRef>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "#DCDCE9",
        "borderBottomWidth": 11.5,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 6.571428571428571,
        "borderRadius": 2,
        "borderRightColor": "transparent",
        "borderRightWidth": 6.571428571428571,
        "borderStyle": "solid",
        "bottom": 40.25,
        "height": 0,
        "left": 70.64285714285714,
        "position": "absolute",
        "transform": Array [
          Object {
            "rotate": "19deg",
          },
        ],
        "width": 0,
      }
    }
  />
  <ForwardRef
    activeOpacity={1}
    style={
      Object {
        "aspectRatio": 1,
        "backgroundColor": "#DCDCE9",
        "borderRadius": 100,
        "height": 59.14285714285714,
        "position": "absolute",
        "top": 115,
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "#DCDCE9",
          "borderRadius": 100,
          "borderWidth": 1,
          "flex": 1,
          "margin": 2,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "aspectRatio": 1,
            "borderRadius": 100,
            "height": "100%",
            "tintColor": "grey",
          }
        }
      />
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 100,
            "height": "100%",
            "opacity": 0.8,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
    </View>
  </ForwardRef>
  <Text
    style={
      Object {
        "color": "#8F8FAF",
        "fontFamily": "BananaGrotesk-Regular",
        "fontSize": 23,
        "letterSpacing": -0.08,
      }
    }
  >
    Carlota
  </Text>
</ForwardRef>
`;
